code/object/objcollide.cpp

```diff
@@ -607,23 +608,21 @@ void obj_reset_colliders()
 
 void obj_collide_retime_cached_pairs()
 {
-	for ( auto& pair : Collision_cached_pairs ) {
-		pair.second.next_check_time = timestamp(0);
+	auto it = Collision_cached_pairs.begin();
+	while (it != Collision_cached_pairs.end()) {
+		auto &pair = it->second;
+		if (pair.signature_a != pair.a->signature || pair.signature_b != pair.b->signature) {
```

Comment:
This is true, they are not directly related, but it's not worth doing on it's own, and we're here anyway, so it might as well be done when they must be iterated over regardless. And they only "should" be removed in the first place, because this very loop is slowed down by them, invalid pairs does not really impinge on normal collision evaluation.

Reply:
Ok.